### ISSUE TITLE
Parse Chapter, Specific Audiobook

### ISSUE THREAD

**Issue by audiojunky23** [bookbonobo/libby-download-extension#193]
I've been using Libby and Libby Downloader for about 2 months now and have not run across this issue yet. I'm on a Windows device. When I try to download The Restaurant at the End of the Universe by Douglas Adams, with Parse Chapters On, I end up with the following error:

![image](https://github.com/bookbonobo/libby-download-extension/assets/166965228/219f5a31-51cc-4749-9666-687cdceadb75)

This is the case whether Precise Chapters is On or Off. If Parse Chapters is Off, then it will process and download regardless of Precise Chapters being On or Off. Since I normally have both turned on for my books, I was wondering what can be done to replicate the format of the other audiobooks I have used ie a single MP3 file with a CUE file rather than multiple MP3 files.

**Comment by mkgtu** [comment]
You say "specific audiobook". Does that mean the the process works ok with other books? Maybe just an issue with this book?

**Comment by audiojunky23** [comment]
Yeah, for any other audiobook I can have Parse Chapters and Precise Chapters On (or any combination of Off and On) save for this one. It can be downloaded with Parse Chapters Off and with Precise Chapters On or Off, but not with Parse Chapters On. I'm only posting here because of my lack of understanding of the software, it does not make sense to me why otherwise the audiobook could be downloaded with other settings.

**Comment by mkgtu** [comment]
> was wondering what can be done to replicate the format of the other audiobooks I have used ie a single MP3 file with a CUE file rather than multiple MP3 files

Since this particular book today became available to borrow at one of my libraries, I borrowed it and experimented. 

With this Firefox extension I got the same result as you. Parse on: failure. Parse off+  precise on: one mp3 file, no cue file. 

The problem is with the chapters (or lack of them) on the  Libby version of this  particular book. I easily downloaded the same  book using the Chrome extension Libby Audiobook Exporter and also got the book with no chapters. However the corresponding Kindle ebook does have 34 chapters! And if I download the same book using the old Overdrive app method, the result is an audiobook with 34 chapters! (The Libby and Overdrive versions of books are not always the same.) In the Overdrive app the chapters of this  book are frequently split into parts. But if you play the Overdrive version in Smart Audiobook Player or Listen Audiobook Player (both android), those chapter splits are neatly cleaned up. 

A few observations. 

FIRST: If you really want to get these books in the shape you want, you would do well to acquaint yourself with alternative download methods. 

SECOND I don't know what you use to play your books. I use Listen Audiobook Player on my android phone, and I also have Smart Audiobook Player. Both of these can expertly handle any format of audiobook thrown at them. They are both excellent. Most anything else on android looks and functions like junk in comparison IMO. On Apple products there are some good options, but none that I know of that can display chapters from all these different formats. Kind of hit and miss.

THIRD I don't understand your concern about single mp3 vs multiple mp3. What's the difference, really? The book size is the same with both, the files (single or multiple) need to be contained in a book folder for most players to recognize them as books. The Firefox extension creates a single mp3 plus cue, the Chrome extension multiple mp3's plus a json file for metadata, the Overdrive app uses multiple mp3's plus a cover image (the metadata is contained in the individual mp3 files). 

Going forward, with possible extension glitches, etc, options may become your best friends. 

This is the Overdrive (OD) version of the book. You'll need one of the two android players above to display the chapters. I think Bound audiobook player on iOS will display these chapters also. (Of course the book will "Play" in any player. 

https://www.dropbox.com/scl/fo/xa7jnsijmeocjk1aoxplo/APffPArKf-V25HoknNK6Sgc?rlkey=4tdl86bd5ew9v1gmqnp4wreci&dl=0
